specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json

```diff
@@ -692,6 +756,65 @@
     }
   },
   "definitions": {
+    "ApplicationGatewayOnDemandProbe": {
+      "properties": {
+        "protocol": {
+          "type": "string",
+          "description": "The protocol used for the probe. Possible values are 'Http' and 'Https'.",
+          "enum": [
+            "Http",
+            "Https"
+          ],
+          "x-ms-enum": {
+            "name": "ApplicationGatewayProtocol",
+            "modelAsString": true
+          }
+        },
+        "host": {
+          "type": "string",
+          "description": "Host name to send the probe to."
+        },
+        "path": {
+          "type": "string",
+          "description": "Relative path of probe. Valid path starts from '/'. Probe is sent to <Protocol>://<host>:<port><path>"
+        },
+        "timeout": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The probe timeout in seconds. Probe marked as failed if valid response is not received with this timeout period. Acceptable values are from 1 second to 86400 seconds."
+        },
+        "pickHostNameFromBackendHttpSettings": {
```

Comment:
As a general rule, a string enum is more future proof, either by modeling as enum with values "true" or "false" or refactoring to be: hostNameSourc with enum "backendHTTPSettings" or "default"

Reply:
would like to it keep same as an exisiting definition (ApplicationGatewayProbePropertiesFormat (line 1602)) for reuse